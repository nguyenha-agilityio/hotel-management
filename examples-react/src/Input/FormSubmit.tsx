import React from "react";

export default function FormSubmit() {
  function handleSubmit(e: React.FormEvent<HTMLFormElement>) {
    e.preventDefault();

    const form = e.target as HTMLFormElement;

    const formData = new FormData(form);

    const formJson = Object.fromEntries(formData.entries());
    console.log(formJson);
  }
  return (
    <form onSubmit={handleSubmit}>
      <h1>Info</h1>
      <label>
        My name is <input name="myInput" defaultValue={"VietDuong"} />
      </label>
      <label>
        Choose <input type="checkbox" name="myCheckBox" defaultChecked={true} />
      </label>
      <p>
        Your favorites
        <label>
          <input type="radio" name="myRadio" value="option1" />
          swimming
        </label>
        <label>
          <input
            type="radio"
            name="myRadio"
            value="option2"
            defaultChecked={true}
          />
          play soccer
        </label>
        <label>
          <input type="radio" name="myRadio" value="option3" />
          dancing
        </label>
      </p>
      <button type="reset">Reset Form</button>
      <button type="submit">Submit Form</button>
    </form>
  );
}
