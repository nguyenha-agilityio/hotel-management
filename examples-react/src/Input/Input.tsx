export default function Input() {
  return (
    <>
      <label>
        My name is <input name="myInput" />
      </label>
      <hr />
      <label>
        choose <input type="checkbox" name="myCheckBox" />
      </label>
      <p>
        Button Radio
        <label>
          <input type="radio" name="myRadio" id="" value="option1" />
          Option1
        </label>
        <label>
          <input type="radio" name="myRadio" id="" value="option2" />
          Option2
        </label>
        <label>
          <input type="radio" name="myRadio" id="" value="option3" />
          Option3
        </label>
      </p>
    </>
  );
}
