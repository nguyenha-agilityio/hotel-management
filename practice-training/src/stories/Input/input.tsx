import { ChangeEventHandler } from "react";

export interface InputProps {
  label: string;
  placeholder: string;
  type: string;
  value: string | number;
  backgroundColor: string;
  size: "md" | "lg" | "sm";
  borderRadius: number;
  color: string;
  onChange: ChangeEventHandler<HTMLInputElement>;
  direction: "row" | "column" | "row-reverse" | "column-reverse";
}

export default function Input({
  label,
  placeholder = "Text",
  type,
  value,
  backgroundColor,
  size = "lg",
  borderRadius,
  color,
  onChange,
  direction = "column",
}: InputProps) {
  let padding = "10px";
  if (size === "sm") padding = "5px";
  if (size === "lg") padding = "15px";
  const style = {
    display: "flex",
    flexDirection: direction,
    backgroundColor,
    padding,
    borderRadius: `${borderRadius}px`,
    color,
    border: "1px solid #ccc",
  };
  return (
    <div>
      <label>{label}</label>
      <input
        type={type}
        placeholder={placeholder}
        value={value}
        onChange={onChange}
        style={style}
      />
    </div>
  );
}
