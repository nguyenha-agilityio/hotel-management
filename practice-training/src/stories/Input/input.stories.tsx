import Input, { InputProps } from "./input";
import { Meta, StoryFn } from "@storybook/react";

export default {
  title: "Components/Input",
  Component: Input,
  argTypes: {
    handleChange: {
      action: "handleChange",
      direction: {
        control: {
          type: "select",
          options: ["row", "column", "row-reverse", "column-reverse"],
        },
      },
      borderRadius: { control: { type: "number" } },
      color: { control: { type: "string" } },
    },
  },
} as Meta<typeof Input>;

const Template: StoryFn<InputProps> = (args: InputProps) => <Input {...args} />;

export const Default = Template.bind({});
Default.args = {
  direction: "column",
  label: "Email",
  backgroundColor: "White",
  size: "md",
  borderRadius: 5,
  color: "black",
};

export const SizeSmall = Template.bind({});
SizeSmall.args = {
  direction: "column",
  label: "Email",
  backgroundColor: "White",
  size: "sm",
  borderRadius: 5,
  color: "black",
};
