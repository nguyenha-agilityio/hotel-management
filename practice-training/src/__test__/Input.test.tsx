import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Input from "../Components/Input/input";
import "@testing-library/jest-dom/extend-expect";

describe("Input Component", () => {
  const mockOnChange = jest.fn();

  test("renders correctly with label and default placeholder", () => {
    const { getByLabelText, getByPlaceholderText } = render(
      <Input label="Test Label" type="text" value="" onChange={mockOnChange} />,
    );

    expect(getByLabelText("Test Label")).toBeInTheDocument();
    expect(getByPlaceholderText("Text")).toBeInTheDocument();
  });

  test("renders with provided placeholder", () => {
    const { getByPlaceholderText } = render(
      <Input
        label="Test Label"
        type="text"
        value=""
        onChange={mockOnChange}
        placeholder="Custom Placeholder"
      />,
    );

    expect(getByPlaceholderText("Custom Placeholder")).toBeInTheDocument();
  });

  test("calls onChange handler when value changes", () => {
    const { getByLabelText } = render(
      <Input label="Test Label" type="text" value="" onChange={mockOnChange} />,
    );

    fireEvent.change(getByLabelText("Test Label"), {
      target: { value: "New Value" },
    });
    expect(mockOnChange).toHaveBeenCalledWith(
      expect.objectContaining({ target: { value: "New Value" } }),
    );
  });

  test("applies correct styles based on props", () => {
    const { getByLabelText } = render(
      <Input
        label="Test Label"
        type="text"
        value=""
        onChange={mockOnChange}
        backgroundColor="lightblue"
        size="sm"
        borderRadius={10}
        color="red"
        direction="row"
      />,
    );

    const inputElement = getByLabelText("Test Label")
      .nextSibling as HTMLInputElement;

    expect(inputElement).toHaveStyle("background-color: lightblue");
    expect(inputElement).toHaveStyle("padding: 5px"); // For size "sm"
    expect(inputElement).toHaveStyle("border-radius: 10px");
    expect(inputElement).toHaveStyle("color: red");
    expect(inputElement.parentElement).toHaveStyle("flex-direction: row");
  });

  test("Matches the snapshot", () => {
    const { asFragment } = render(
      <Input
        label="Test label"
        type="text"
        value=" "
        onChange={mockOnChange}
        color="white"
        direction="row"
        borderRadius={5}
        size="md"
        backgroundColor="black"
      />,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
