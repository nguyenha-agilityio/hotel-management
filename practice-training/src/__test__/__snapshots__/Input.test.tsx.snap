// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component Matches the snapshot 1`] = `
<DocumentFragment>
  <div>
    <label>
      Test label
    </label>
    <input
      placeholder="Text"
      style="display: flex; flex-direction: row; background-color: black; padding: 10px; border-radius: 5px; color: white; border: 1px solid #ccc;"
      type="text"
      value=" "
    />
  </div>
</DocumentFragment>
`;
